src/Groups/perm.jl

```diff
@@ -387,3 +387,168 @@ function cycle_structure(g::PermGroupElem)
+        if typeof(ex) != Expr
+            error("Input is not a permutation.")
+        end
+        if ex.head != :call && ex.head != :tuple
+            error("Input is not a permutation.")
+        end
+    end
+    
+    if ex.head != :tuple
+        error("Input is not a permutation.")
+    end
+    
+    pushfirst!(res, Expr(:vect,ex.args...))
+
+    return esc(:(Oscar.cperm($(res...))))
+end
+
+
+################################################################################
+#
+#   perm(n,gens)
+#
+@doc Markdown.doc"""
+    @perm(n,gens)
+Macro to input a list of permutations which are generated as elements of
+the `symmetric_group(n)` with the function `cperm`.
+# Examples
+```jldoctest
+julia> gens = Oscar.Permutations.@perm 14 [
+               (1,10)
+              (2,11)
+              (3,12)
+              (4,13)
+              (5,14)
+              (6,8)
+              (7,9)
+              (1,2,3,4,5,6,7)(8,9,10,11,12,13,14)
+              (1,2)(10,11)
+             ]
+9-element Vector{PermGroupElem}:
+ (1,10)
+ (2,11)
+ (3,12)
+ (4,13)
+ (5,14)
+ (6,8)
+ (7,9)
+ (1,2,3,4,5,6,7)(8,9,10,11,12,13,14)
+ (1,2)(10,11)
+julia> gens[1].parent
+Sym( [ 1 .. 14 ] )
+```
+"""
+macro perm(n,gens)
+
+    s = symmetric_group(n)
+    ores = Vector{Expr}(undef,length(gens.args))
+    i = 1
+    for ex in gens.args
+        res = []
+        
+        if typeof(ex) != Expr
+            throw(ArgumentError("Input is not a permutation."))
+            error("Input is not a permutation.")
+        end
+        
+        if ex.head != :call && ex.head != :tuple
+            error("Input is not a permutation.")
+        end
+       
+        while ex.head == :call
+            pushfirst!(res, Expr(:vect, ex.args[2:end]...))
+            ex = ex.args[1]
+            if typeof(ex) != Expr
+                error("Input is not a permutation.")
+            end
+            if ex.head != :call && ex.head != :tuple
+                error("Input is not a permutation.")
+            end
+        end
+        
+        if ex.head != :tuple
+            error("Input is not a permutation.")
+        end
+        
+        pushfirst!(res, Expr(:vect,ex.args...))
+
+        ores[i] = esc(:(Oscar.cperm(symmetric_group($n),$(res...))))
+        i = i + 1
+    end
+
+    return Expr(:vect,ores...)
+end
+
+
+################################################################################
+#
+#   permgroup(n::Int64,gens::Vector{PermGroupElem})
+#
+@doc Markdown.doc"""
+    permgroup(n::Int64,gens::Vector{PermGroupElem})
+Generates a `PermGroup` with generators gens as a subgroup of `symmetric_group(n)`.
+# Examples
+```jldoctest
+julia> gens = Oscar.Permutations.@perm 14 [
+               (1,10)
+              (2,11)
+              (3,12)
+              (4,13)
+              (5,14)
+              (6,8)
+              (7,9)
+              (1,2,3,4,5,6,7)(8,9,10,11,12,13,14)
+              (1,2)(10,11)
+             ];
+julia> G = Oscar.Permutations.permgroup(14,gens)
```

Comment:
I don't see why this is needed? Just write
```suggestion
julia> G = sub(symmetric_group(14),gens)[1]
```